src/librustdoc/json/conversions.rs

```diff
@@ -119,11 +119,11 @@ impl FromWithTcx<clean::GenericArgs> for GenericArgs {
         use clean::GenericArgs::*;
         match args {
             AngleBracketed { args, bindings } => GenericArgs::AngleBracketed {
-                args: args.into_iter().map(|a| a.into_tcx(tcx)).collect(),
+                args: args.into_vec().into_iter().map(|a| a.into_tcx(tcx)).collect(),
```

Comment:
Why do you need the `.into_vec()` here? Shouldn't `.into_iter()` be enough for a boxed slice?

Reply:
Do you think it would be worth it to use `Box::into_inner` then? (it's unstable and I'm not sure about the perf)

Reply:
Just tried:

```rust
let x: Box<[String]> = Box::new(["a".to_owned()]);
for a in x.into_iter() {
    println!("-> {}", a);
}
```

and it works fine. I think I'm missing something here so I'll give it a try tomorrow locally to try to understand.

Reply:
Ok, that was the missing part. Thanks for the explanations!